### PR TITLE
LON-962

### PR DESCRIPTION
**Pull request by SandraPG85**
+ Small fix for issue 965
+ Tests fixed with some minor code changes required (some tests are still failing but pending to be checked with Andoni)

### CODE REVIEW

**Comment by vguzmanp** [comment]
The branch should be called LON-962, to make JIRA link discover the branches & PR automatically.

**Comment by vguzmanp** [comment]
To avoid github's truncating of the commit messages, we follow (or try to follow) [these guidelines](https://fluendo.atlassian.net/wiki/display/TEAM/Commit+message+style)

File: Tests/Controller/TestProjectAnalysisController.cs [hunk 1]
```diff
@@ -133,55 +135,16 @@ public void TearDown ()
...
-			App.Current.EventsBroker.Publish<OpenNewProjectEvent> (
-				new OpenNewProjectEvent {
-					Project = project,
-					ProjectType = ProjectType.CaptureProject,
-					CaptureSettings = settings
-				}
-			);
-			Assert.AreEqual (1, projectOpened);
-			testedProject = null;
+            testedProject = null;
```
Comment (vguzmanp) [inline Tests/Controller/TestProjectAnalysisController.cs]: `testedProject`is not used

File: Tests/Controller/TestProjectAnalysisController.cs [hunk 2]
```diff
@@ -191,51 +154,47 @@ public void TestCaptureProjectError ()
 			/* Errors during a capture project should be handled gracefully
 			 * closing the current capture project and saving a copy of the
 			 * captured video and coded data, without loosing anything */
-			Assert.AreEqual (1, App.Current.DatabaseManager.ActiveDB.Count<LMProject> ());
-			Assert.AreEqual (2, projectOpened);
-
-			App.Current.EventsBroker.Unsubscribe<OpenedProjectEvent> (et);
+			// SP-Question: The comment does not follow the actual code implementation
+			// The project is not saved, what is the expected behavior?
```
Comment (vguzmanp) [inline Tests/Controller/TestProjectAnalysisController.cs]: I think it should. Someone should be subscribed to `CaptureErrorEvent` and save the project.

File: Tests/Controller/TestProjectAnalysisController.cs [hunk 3]
```diff
@@ -98,6 +99,7 @@ public void FixtureSetup ()
 			player = new VideoPlayerController ();
 			videoPlayerVM = new VideoPlayerVM ();
 			videoPlayerVM.Player = player;
+			player.SetViewModel (videoPlayerVM);
```
Comment (vguzmanp) [inline Tests/Controller/TestProjectAnalysisController.cs]: Why was this needed?

File: LongoMatch.Drawing/CanvasObjects/Location/TimelineEventLocationView.cs [hunk 4]
```diff
@@ -87,6 +87,7 @@ void UpdatePoints ()
 		void HandleViewModelPropertyChanged (object sender, PropertyChangedEventArgs e)
 		{
 			// FIXME: Add positions to the VM
+			// SP-Question: sender will be always the viewmodel so the condition will be never true
```
Comment (vguzmanp) [inline LongoMatch.Drawing/CanvasObjects/Location/TimelineEventLocationView.cs]: Right now, if the Model of a ViewModel changes one of the properties, it will send a PropertyChanged event with the sender set to itself. The ViewModel will receive this event and forward it as is, without changing the sender.
So right now this code will be called "correctly".

I think we had a task in the backlog to change the sender when forwarding.

File: Tests/Controller/TestProjectAnalysisController.cs (hunk 3 above)
Comment (SandraPG85) [inline Tests/Controller/TestProjectAnalysisController.cs]: the setviewmodel is called by the state since we are not using in the test the MoveTo it has to be done manually, otherwise the view model is not assigned.

File: Tests/Controller/TestProjectAnalysisController.cs (hunk 2 above)
Comment (SandraPG85) [inline Tests/Controller/TestProjectAnalysisController.cs]: The ProjectAnalysisController is already subscribed to the event and it does the same call that the CaptureFinishedEvent the only difference is that the save operation seems to be forced to be skipped. So my question is if this is or not on purpose.

File: Tests/Controller/TestProjectAnalysisController.cs (hunk 1 above)
Comment (SandraPG85) [inline Tests/Controller/TestProjectAnalysisController.cs]: fix

File: Tests/Controller/TestProjectAnalysisController.cs [hunk 5]
```diff
@@ -36,6 +36,7 @@
 using LongoMatch.Core.ViewModel;
 using LongoMatch.Services.ViewModel;
 using VAS.Core.ViewModel;
+using LongoMatch.Services.State;
```
Comment (SandraPG85) [inline Tests/Controller/TestProjectAnalysisController.cs]: fix

File: Tests/Controller/TestProjectAnalysisController.cs [hunk 6]
```diff
@@ -260,52 +219,35 @@ public void TestCaptureFinished ()
...
+			Assert.AreEqual (project, projectsManager.Project.Model);
 			Assert.AreEqual (1, App.Current.DatabaseManager.ActiveDB.Count<LMProject> ());
-			Assert.AreEqual (2, projectChanged);
-
-			App.Current.EventsBroker.Unsubscribe<OpenedProjectEvent> (et);
 		}
 
-		[Test ()]
+		// SP-Question: Has sense this test in here after latest changes?
+		/*[Test ()]
```
Comment (SandraPG85) [inline Tests/Controller/TestProjectAnalysisController.cs]: Test enabled again but it failing, I will fix it after discussing with Andoni the Open behaviour.

File: Tests/Drawing/Objects/TestTimelineEventLocationView.cs [hunk 7]
```diff
@@ -74,12 +74,9 @@ public void TestEventDrawWhenPointChanges ()
 				BackgroundWidth = 100,
 				BackgroundHeight = 100,
 			};
-			view.SetViewModel (timelineEventVM);
-			view.RedrawEvent += (co, area) => redrawCount++;
 
 			timelineEvent.EventType.TagFieldPosition = true;
 			timelineEvent.AddDefaultPositions ();
-			view.Draw (tkMock.Object, null);
```
Comment (SandraPG85) [inline Tests/Drawing/Objects/TestTimelineEventLocationView.cs]: changes on test reverted, it is failing because the redraw is done by the notify property changed of the view model (pending to clarify this part with Andoni)

File: LongoMatch.Drawing/CanvasObjects/Location/TimelineEventLocationView.cs (hunk 4 above)
Comment (SandraPG85) [inline LongoMatch.Drawing/CanvasObjects/Location/TimelineEventLocationView.cs]: We have talk about it, I understand better now how it is working but I still think that also changes of the view model should be able to force a redraw.
Pending to discuss it further.
